### ISSUE TITLE
Triply Use Case 3: Finding and applying a 3D CRS

### ISSUE THREAD

**Issue by wouterbeek** [opengeospatial/ogc-geosparql#585]
# Triply Use Case 3: Finding and applying a 3D CRS

## Description

In GeoSPARQL, I can specify the CRS for my geometries in GML tags or at the start of WKT literals. It is not always clear which CRS IRI I should use. I can look up a CRS in <https://epsg.org>, but this gives me IRIs like <https://epsg.org/crs_28992/Amersfoort-RD-New.html?sessionkey=7gzojo4hdw> that may not be understood by other triple stores. For 3D data, it is difficult to find a CRS that applies. Many CRSes include descriptions like "Cartesian 2D CS. Axes: easting, northing (X,Y). Orientations: east, north. UoM: m.Open" that seem to indicate that they cannot be used for 3D data. Where should I go to find a 3D CRS? If I cannot find a 3D CRS for my locale, should I first convert to some other CRS? Is <https://epsg.org> a / the authoritative source in the first place? Etc.

## Actor

- Data Expert (who may not be a Geodata Expert)

## Preconditions

- I have 3D data, but not yet in GeoSPARQL.
- I know how to store 2D data in GeoSPARQL.
- I have not found a CRS that works for my 3D data.

## Postconditions

- I have stored all 3D data as GeoSPARQL in my triple store, with a CRS IRI that supports 3D and is recognized by other triple stores.

## Steps

1. I have 3D data about buildings in The Netherlands.
2. I have 2D data in GeoSPARQL for which I use CRS IRI <http://www.opengis.net/def/crs/EPSG/0/28992>, which is specific to The Netherlands.
3. I do not know whether I can also use this CRS IRI for 3D data, so I query an authoritative source (preferably using SPARQL, but otherwise in some other way) to inquire (a) whether the CRS supports 3D, (b) in case the CRS does not support 3D, which alternative CRS would support 3D.
4. Once I have found a CRS IRI that supports 3D in The Netherlands, I integrate all my data into one GeoSPARQL dataset. All GeoSPARQL literals in my dataset include the correct CRS IRI.
5. When somebody else queries my dataset, they are able to interpret my CRS IRI.

**Comment by FransKnibbe** [comment]
I read this use case as illustrating the need for a CRS ontology, and through that, an official Linked Data bases CRS registry. This is the topic of https://github.com/opengeospatial/ontology-crs. 3D CRSs do exist. For official data in the Netherlands, that would be [RD+NAP](https://epsg.org/crs_7415/Amersfoort-RD-New-NAP-height.html), with its OGC IRI being http://www.opengis.net/def/crs/EPSG/0/7415. But if you don't know this beforehand, it is difficult to find applicable 3D CRSs for 3D models.

Another aspect is applying the CRS IRI, once it is found. I think that does call for a inCRS or hasCRS property (see issue #82 and  issue #419). For 3D geometry this is especially needed, because 3D geometry formats typically are not geographical, and have no built-in way of referencing a CRS.

**Comment by situx** [comment]
Agreed. This is a CRS ontology need and pertains to the issues @FransKnibbe mentioned

**Comment by wouterbeek** [comment]
@FransKnibbe and @situx Thank you for your feedback. I have now split this issue into _applying_ a 3D CRS IRI (this issue), and _finding_ a CRS IRI (issue #599). Let me know in case I can make additional improvements to either issue.
